pytorch_lightning/trainer/trainer.py

```diff
@@ -746,24 +746,29 @@ def fit(self, model):
             trainer.fit()
         """
         # when using multi-node or DDP within a node start each module in a separate process
+        output = 1
         if self.use_ddp2:
+            self.model = model
             task = int(os.environ['SLURM_LOCALID'])
-            self.ddp_train(task, model)
+            output = self.ddp_train(task, model)
 
         elif self.use_ddp:
+            self.model = model
             if self.is_slurm_managing_tasks:
                 task = int(os.environ['SLURM_LOCALID'])
-                self.ddp_train(task, model)
+                output = self.ddp_train(task, model)
             else:
                 mp.spawn(self.ddp_train, nprocs=self.num_gpus, args=(model,))
 
         # 1 gpu or dp option triggers training using DP module
         # easier to avoid NCCL issues
         elif self.use_dp:
-            self.dp_train(model)
+            self.model = model
```

Comment:
I would add log message that you are switching from an already assigned model to the newly passed model
for a user, it may simplify the debugging why the model has another performances, etc.